test/scripts/init.js

```diff
@@ -31,41 +32,58 @@ describe('init', () => {
     });
   }
 
-  function compareFile(a, b) {
+  async function compareFile(a, b) {
     const streamA = createSha1Hash();
     const streamB = createSha1Hash();
 
-    return Promise.all([
-      pipeStream(fs.createReadStream(a), streamA),
-      pipeStream(fs.createReadStream(b), streamB)
-    ]).then(() => streamA.read().equals(streamB.read()));
+    await Promise.all([
+      pipeStream(createReadStream(a), streamA),
+      pipeStream(createReadStream(b), streamB)
+    ]);
+
+    return streamA.read().equals(streamB.read());
   }
 
-  function check(path) {
-    return Promise.each(assets, item => compareFile(
-      pathFn.join(assetDir, item),
-      pathFn.join(path, item)
-    ).should.eventually.be.true).finally(() => rmdir(path));
+  async function check(path) {
+    for (const item of assets) {
+      const result = await compareFile(
+        join(assetDir, item),
+        join(path, item)
+      );
+
+      result.should.be.true;
+    }
+
   }
 
   function withoutSpawn(fn) {
     return initModule.__with__('spawn', () => Promise.reject(new Error('spawn is not available')))(fn);
   }
 
-  before(() => fs.listDir(assetDir).then(files => {
+  before(async () => {
+    const files = await listDir(assetDir);
     assets = files;
-  }));
+  });
 
-  after(() => rmdir(baseDir));
+  after(async () => await rmDir(baseDir));
 
-  it('current path', () => withoutSpawn(() => init({_: []}).then(() => check(baseDir))));
+  it('current path', () => withoutSpawn(async () => {
+    await init({_: []});
+    await check(baseDir);
+  }));
 
-  it('relative path', () => withoutSpawn(() => init({_: ['test']}).then(() => check(pathFn.join(baseDir, 'test')))));
+  it('relative path', () => withoutSpawn(async () => {
+    await init({_: ['test']});
+    await check(join(baseDir, 'test'));
```

Comment:
It seems the test doesn't work.  Although I change the expect `check(join(baseDir, 'hello'))`, It still is passed due to `assets` array aways empty with spawn mocked.

Reply:
I just tested reverting this commit, it didn't work too.